.github/workflows/ci.yml

```diff
@@ -147,10 +147,15 @@ jobs:
       - name: "Build"
         run: nix develop -L --no-update-lock-file --command just build-cli
 
-      - name: "CLI Tests"
+      - name: "CLI Unit Tests"
         env:
           RUST_BACKTRACE: 1
-        run: nix develop -L --no-update-lock-file --command just test-cli
+        run: nix develop -L --no-update-lock-file --command just impure-tests
+
+      - name: "CLI Integration Tests"
+        env:
+          RUST_BACKTRACE: 1
+        run: nix develop -L --no-update-lock-file --command just integ-tests
```

Comment:
question nonblocking: can we back this out now?

Reply:
I'd prefer this be a separate step either way, instead of running unit tests and integration tests back to back in one step (also separates logs for both)